engine/src/main/java/org/terasology/engine/rendering/nui/layers/ingame/metrics/RunningThreadsMode.java

```diff
@@ -3,25 +3,42 @@
 
 package org.terasology.engine.rendering.nui.layers.ingame.metrics;
 
-import org.terasology.engine.monitoring.ThreadMonitor;
-import org.terasology.engine.monitoring.impl.SingleThreadMonitor;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.search.Search;
+import org.terasology.engine.monitoring.DisplayMetricsMonitor;
+
+import java.util.concurrent.TimeUnit;
 
 final class RunningThreadsMode extends MetricsMode {
 
-     RunningThreadsMode() {
-        super("\n- Running Threads -");
+    RunningThreadsMode() {
+        super("\n- Task Metrics -");
     }
 
     @Override
     public String getMetrics() {
+
         StringBuilder builder = new StringBuilder();
         builder.append(getName());
         builder.append("\n");
-        ThreadMonitor.getThreadMonitors(true).stream().filter(SingleThreadMonitor::isActive).forEach(threads -> {
-            builder.append(threads.getName());
-            builder.append(" - ");
-            builder.append(threads.getLastTask());
-            builder.append("\n");
+
+        Search.in(DisplayMetricsMonitor.metricRegistry)
+                .tag("monitor", "display-metric")
+                .timers().forEach(k -> {
+
+            HistogramSnapshot snapshot = k.takeSnapshot();
```

Comment:
I'm trying to figure out how this compares to the previous implementation.

The old one was a list of all _currently active tasks._

If I'm reading this correctly, the new one is a list of timers that had something recorded in the previous reporting period (DisplayMetricsMonitor.captureDuration), with some stats about throughput and latency.

Those are also useful things to know, but they're not the same.

Is that a problem in practice?

- is there more lag this way? i.e. if it only updates once every captureDuration.
  + is it enough to matter? does anyone _want_ updates more than once per second?
- how often are there tasks with duration greater than captureDuration? If this happens, I'm concerned it gives a less accurate notion of tasks that are still processing.

Reply:
Also: This is searching the DisplayMetricsMonitor registry.

but the ThreadMonitor timers are registered in GlobalRegistry.

Is this search going to find those timers?

Reply:
metric registry is where the data is supposed to get sinked to. its an empty registry since we are not pushing these metrics anywhere.  DismplayMetricMonitor metric registry is a blank registry with a 1 second interval for the sampled data. 

- tells you how long it takes to process an task/ the number of entries that are getting processed 
- The capture is driven by when the entry was recorded i.e the task ending. 

I would probably use JMX if you need a better picture. the snapshots have a lot more data then what I can present within the scope of this ticket.

Reply:
Posted some implementation ideas related to "currently active tasks" in https://github.com/MovingBlocks/Terasology/pull/4799#issuecomment-894694532

Reply:
resolved by declaring micrometer to be out-of-scope for this PR (#4839)